CAP/gap/PrintingFunctions.gi

```diff
@@ -3,133 +3,40 @@
-        for j in i do
-            j[ 2 ] := ReplacedString( j[ 2 ], "abelian", "Abelian" );
-        od;
-    od;
-    
-    for i in [ 1 .. Length( return_list ) ] do
-        if ForAny( return_list[ i ], j -> j[2] = "Abelian" ) then
-            return_list := Concatenation( return_list{[ 1 .. i - 1 ]}, return_list{[ i + 1 .. Length( return_list ) ]}, [ return_list[ i ] ] );
-        fi;
-    od;
-    
-    return return_list;
-    
-end );
-
-CAP_INTERNAL_CATEGORY_PROPERTY_RANK_AND_STRING.sorted_list := __CAP_INTERNAL_RANK_CATEGORY_FROM_METHOD_RECORD();
-
 InstallGlobalFunction( InfoStringOfInstalledOperationsOfCategory,
   
   function( category )
-    local prim_operations, derived_operations, adjectives, current_stack, current_property, current_adjective, string;
+    local prim_operations, derived_operations, list_of_properties, property, string;
     
     if not IsCapCategory( category ) then
         Error( "first argument must be a category" );
         return;
     fi;
     
-    prim_operations := ListPrimitivelyInstalledOperationsOfCategory( category );
-    derived_operations := ListInstalledOperationsOfCategory( category );
+    prim_operations := Length( ListPrimitivelyInstalledOperationsOfCategory( category ) );
+    derived_operations := Length( ListInstalledOperationsOfCategory( category ) );
     
-    prim_operations := Length( prim_operations );
-    derived_operations := Length( derived_operations );
+    list_of_properties := ShallowCopy( RecNames( CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD ) );
     
-    adjectives := [ ];
+    Remove( list_of_properties, Position( list_of_properties, "EveryCategory" ) );
     
-    for current_stack in __CAP_INTERNAL_RANK_CATEGORY_FROM_METHOD_RECORD() do
-        for current_property in current_stack do
-            if Length( CheckConstructivenessOfCategory( category, current_property[ 1 ] ) ) = 0 then
-                Add( adjectives, current_property[ 1 ] );
-                break;
-            fi;
-        od;
-    od;
+    list_of_properties := Filtered( list_of_properties, p -> CheckConstructivenessOfCategory( category, p ) = [ ] );
+    
+    list_of_properties := MaximalObjects( list_of_properties,
+                                  { p1, p2 } -> IsSubset( CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.( p2 ), CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.( p1 ) ) );
+    
+    Sort( list_of_properties, { p1, p2 } -> Length( CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.( p2 ) ) > CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.( p1 ) );
     
     string := ShallowCopy( String( prim_operations ) );
     Append( string, " primitive operations were used to derive " );
     Append( string, String( derived_operations ) );
-    Append( string, " operations for this category which" );
-    for current_adjective in adjectives do
+    Append( string, " operations for this category" );
+    if not IsEmpty( list_of_properties ) then
+        Append( string, " which" );
```

Comment:
```suggestion
        Append( string, " which constructively" );
```

to make clear that the displayed properties might differ from `ListKnownCategoricalProperties`.

Reply:
Done.